compiler/qsc_rca/tests/intrinsics.rs

```diff
@@ -0,0 +1,1289 @@
+        "IntAsBigInt",
+        &expect![
+            r#"
+            Callable: CallableComputeProperties:
+                body: ApplicationsGeneratorSet:
+                    inherent: Classical
+                    dynamic_param_applications:
+                        [0]: [Parameter Type Element] Quantum: QuantumProperties:
+                            runtime_features: RuntimeFeatureFlags(UseOfDynamicInt)
+                            value_kind: Element(Dynamic)
+                adj: <none>
+                ctl: <none>
+                ctl-adj: <none>"#
+        ],
+    );
+}
+
+#[test]
+fn check_rca_for_dump_machine() {
+    let compilation_context = CompilationContext::new();
+    check_callable_compute_properties(
+        &compilation_context.fir_store,
+        compilation_context.get_compute_properties(),
+        "DumpMachine",
+        &expect![
+            r#"
+            Callable: CallableComputeProperties:
+                body: ApplicationsGeneratorSet:
+                    inherent: Classical
+                    dynamic_param_applications: <empty>
+                adj: <none>
+                ctl: <none>
+                ctl-adj: <none>"#
+        ],
+    );
+}
+
+#[test]
+fn check_rca_for_check_zero() {
+    let compilation_context = CompilationContext::new();
+    check_callable_compute_properties(
+        &compilation_context.fir_store,
+        compilation_context.get_compute_properties(),
+        "CheckZero",
+        &expect![
+            r#"
+            Callable: CallableComputeProperties:
+                body: ApplicationsGeneratorSet:
+                    inherent: Quantum: QuantumProperties:
+                        runtime_features: RuntimeFeatureFlags(0x0)
+                        value_kind: Element(Dynamic)
+                    dynamic_param_applications:
+                        [0]: [Parameter Type Element] Quantum: QuantumProperties:
+                            runtime_features: RuntimeFeatureFlags(UseOfDynamicQubit)
+                            value_kind: Element(Dynamic)
+                adj: <none>
+                ctl: <none>
+                ctl-adj: <none>"#
+        ],
+    );
+}
+
+#[test]
+fn check_rca_for_message() {
+    let compilation_context = CompilationContext::new();
+    check_callable_compute_properties(
+        &compilation_context.fir_store,
+        compilation_context.get_compute_properties(),
+        "Message",
+        &expect![
+            r#"
+            Callable: CallableComputeProperties:
+                body: ApplicationsGeneratorSet:
+                    inherent: Classical
+                    dynamic_param_applications:
+                        [0]: [Parameter Type Element] Quantum: QuantumProperties:
+                            runtime_features: RuntimeFeatureFlags(UseOfDynamicString)
+                            value_kind: Element(Static)
+                adj: <none>
+                ctl: <none>
+                ctl-adj: <none>"#
+        ],
+    );
+}
```

Comment:
Future: Hmm, this brings to mind something that was a bit of a debate in the old stack: should dynamic strings be rejected or ignored? I tend to be of the opinion that asking a user to comment out their strings used for local printf-debugging is not a huge burden, but others have brought up the desire to let folks write code with debug `Message` calls and string interpolation of dynamic values and insisted that the compiler should know to ignore these during analysis for code generation. I don't think it should affect this PR, just wanted to note for the future that a similar complaint might come up if we treat anything that uses dynamic strings as disallowed...

Reply:
Even if strings are mostly used with `Message`, I think the safest thing to do is to reject dynamic strings because otherwise we could have something like this:

```qsharp
use q = Qubit();
let s = if M(q) == Zero { "Foo" } else { "Bar" };
if s == "Foo" {
    X(q);
}
```

I don't think people are doing this, but that doesn't mean they can't.